cEP-0020.md

```diff
@@ -0,0 +1,532 @@
+its processes.
+
+## Why Gamification?
+
+As coala tries to be a welcoming organization for newcomers,
+it gives them clear pathways and as much direct assistance as
+possible. However, it is already known that newcomers face many
+barriers while attempting to contribute to open source for the
+first time, and additional barriers while attempting to
+contribute to projects run by the coala organization.
+
+Some barriers they face include orientation issues that can
+potentially demotivate newcomers from placing their first
+contribution. On the other hand, gamification is widely used
+to engage and motivate people to accomplish tasks and improve
+their performance.
+
+Therefore, the goal of this project is to use gamification to
+orient and motivate newcomers to overcome onboarding barriers
+to contribute and engage with coala and its community.
+
+## Gamification Introduction
+
+There will be four game design elements to help newcomers:
+Activities, Points, Badges and Levels. For each of these
+elements, there will be a set of rules to describe the
+operation of the gamified environment.
+
+### Activities
+
+Let's start by defining some activities which could be tracked:
+
+- Create a bug report
+- Create newcomer difficulty issue
+- Create a bear proposal issue
+- Create a documentation issue
+- Create low difficulty issue
+- Create medium difficulty issue
+- Create pull request that fixes a bug
+- Create pull request for bear proposal
+- Create pull request for a documentation issue
+- Create pull request for newcomer issue
+- Create pull request for low difficulty issue
+- Create pull request for medium difficulty issue
+- Gitmate errors on pull request
+- CI status of the pull request
+- Review comments on pull request
+- Number of questions asked in `coala/coala` room
+- Number of answers given in `coala/coala` room
+- Number of questions asked in `coala-bears` room
+- Number of  answers given in `coala-bears` room
+- Number of questions asked in `coala/corobo` room
+- Number of answers given in `coala/corobo` room
+- Number of questions asked in `coala/community` room
+- Number of answers given in `coala/community` room
+- Number of messages in coala other rooms
+- Make use of `corobo` on Gitter
+- Introduce `.coafile` in user own repository
+- Introduce `.coafile` in an organisation with 50 or more stars
+  for its repositories
+- Introduce CI in the users own repository to run coala using
+  Travis CI
+- Setup the Circle CI to the fork of coala repository
+- Setup Travis CI to the fork of coala repository
+- Setup Appveyor CI to the fork of coala repository
+
+### Points
+
+Points are a simple tool to reward positive contributions
+to coala, and they provide an excellent base upon which to
+build a gamified design. Now let's assign points to some
+of the tracked activities mentioned above:
+
+- Create Issue
+  - 20 points for type/bug
+  - 20 points for bear proposal
+  - 10 points for difficulty newcomer
```

Comment:
To address Makman's point at https://github.com/coala/cEPs/pull/150/files#r190873892 , these points for difficulty level should be decreased, considerably.  creating lots of valid issues shouldnt be an easy way to level up.  creating lots of valid issues should result in a lovely badge.

newcomer issues are usually very tiny problems and very easy to explain because they are so trivial.
low are also very plentiful and are easy to find and write up.
medium issues usually require quite a bit more effort to write up.

maybe 5, 8, 12 ?

also add difficulty high at 15

However, difficulty shouldnt be the way to get a high score for an issue.  *importance* of an issue is more relevant.  Please create additional activities for importance/* labels, and medium importance would be 15, and high would be 20 (i.e. type/bug is usually high importance).  low maybe 10 and trivial = 5 .. ?

Reply:
>Please create additional activities for importance/* .....

Yep, assigning points based on importance is great.

However, one thing to note that we should also check the difficulty of the issue with the importance.
E.g.: An importance/high issue could be difficulty newcomer. And assigning 15 points for this would not be very good?

Reply:
Probably I should add another note saying that:

*There will be a method to check what is the other labels on importance high issues or on other important related issues, and points will be assigned accordingly. E.g.: If an issue is importance high and has difficulty newcomer the suitable points would be `10`.
(points assigned to the important issues are just for showing an example that in these type of situations the importance label will be preferred.)*